### ISSUE TITLE
binary:logistic model selection

### ISSUE THREAD

**Issue by rickyi22** [dmlc/xgboost#2626]
I am trying to using logistic regression to predict the probability of something to happend, so when dealing with train data , i manage to get my y_train labeled in 0 or 1, but when choosing model , i remember that logistic regression is usually for classifier problem, so i am not sure i'll use XGBRegrssor or XGBClassifier. first i was using the later,here's the code
`
clf = xgb.XGBRegressor(max_depth=5, learning_rate=0.1, n_estimators=200, silent=1, objective='binary:logistic', nthread=4)
`
but it went error that y_train should be continuous , so if i am right, i need to set my y_train label between 0 and 1 , so i could use binary:logistic or reg:logistic to output the probability ? and if i use XGBClassifier here , i will just get the output 0 and 1.
new to ML and xgboost , hope someone could help me

**Comment by pommedeterresautee** [comment]
With XGBClassifier you can get the probabilities as you are expecting.
http://xgboost.readthedocs.io/en/latest//parameter.html#parameters-in-r-package
`“binary:logistic” –logistic regression for binary classification, output probability`
